### PR TITLE
Dynamic funding pubkey

### PR DESCRIPTION
**Pull request by pm47**
Best reviewed commit by commit.

A different funding pubkey is used for each splice.

### CODE REVIEW

**Comment by pm47** [comment]
Rebased on master.

File: src/commonTest/kotlin/fr/acinq/lightning/channel/InteractiveTxTestsCommon.kt [hunk 1]
```diff
@@ -1080,18 +1084,22 @@ class InteractiveTxTestsCommon : LightningTestSuite() {
             lockTime: Long
         ): Either<FundingContributionFailure, FundingContributions> {
             val channelId = randomBytes32()
+            val fundingTxIndex = 0L
```
Comment (t-bast) [inline src/commonTest/kotlin/fr/acinq/lightning/channel/InteractiveTxTestsCommon.kt:1087]: Shouldn't this be an argument of `createSpliceContributions`?

File: src/commonTest/kotlin/fr/acinq/lightning/channel/InteractiveTxTestsCommon.kt [hunk 2]
```diff
@@ -1108,26 +1116,31 @@ class InteractiveTxTestsCommon : LightningTestSuite() {
             lockTime: Long
         ): Fixture {
             val channelId = randomBytes32()
+            val fundingTxIndex = 0L
```
Comment (t-bast) [inline src/commonTest/kotlin/fr/acinq/lightning/channel/InteractiveTxTestsCommon.kt:1119]: Same here?

**Review by t-bast — COMMENTED** [review]
Looks mostly good, I have a clean-up commit at https://github.com/ACINQ/lightning-kmp/commit/43e1c1476456fb65c5b5b8a0ec5ee21a719a023f that you may want to cherry-pick

File: src/commonTest/kotlin/fr/acinq/lightning/channel/InteractiveTxTestsCommon.kt (hunk 1 above)
Comment (pm47) [reply src/commonTest/kotlin/fr/acinq/lightning/channel/InteractiveTxTestsCommon.kt]: Done in 4b3df591.

File: src/commonTest/kotlin/fr/acinq/lightning/channel/InteractiveTxTestsCommon.kt (hunk 2 above)
Comment (pm47) [reply src/commonTest/kotlin/fr/acinq/lightning/channel/InteractiveTxTestsCommon.kt:1119]: Done in 4b3df591.

**Comment by pm47** [comment]
> I have a clean-up commit at https://github.com/ACINQ/lightning-kmp/commit/43e1c1476456fb65c5b5b8a0ec5ee21a719a023f that you may want to cherry-pick

Thanks! Applied in [c82ff9e](https://github.com/ACINQ/lightning-kmp/pull/447/commits/c82ff9e4bcf3a2c7443d9f420ea59bd2dc3c29ec).

**Review by t-bast — APPROVED** [review]

**Pull request merged**